sips/sip-012/sip-012-cost-limits-network-upgrade.md

```diff
@@ -0,0 +1,674 @@
+
+Miners, developers, Stackers and community members can demonstrate their support for this network upgrade in one of the following two ways:
+
+* _Send a contract-call transaction to indicate support for the upgrade_: The contract would aggregate the STX balance on the tx-sender account. It could additionally call into the PoX contract to separately aggregate the total Stacked STX amount. See the Appendix for an initial draft of such a contract.
+* _Send a BTC transaction to indicate support_: Many large STX holders are on multi-sig BTC wallets unable to issue contract-calls. Such wallets could instead send a pure BTC transaction to indicate their support for the vote; it would be a no-op for the Stacks chain, but can be verified off-chain see [Bitcoin support indication](#bitcoin-support-indication) for details.
+
+The SIP will be considered Recommended if wallets indicating support for the upgrade (through either mechanism) add up to > 10% of circulating supply of STX (approx 120M).
+
+In terms of how these cost-limits would actually be applied, this SIP proposes the following:
+* Add new functionality to stacks-blockchain that uses the current cost-limits by default and uses new cost-limits if the burn block height exceeds a configurable parameter (could be a compile time configuration to avoid runtime issues)
+* Once a BTC block number has been determined, ship a new stacks-blockchain release at least one week before to give miners and node operators time to upgrade before the upgrade block height is reached
+* In the subsequent release, remove all usage of the old cost-limits and just use the new cost-limits by default
+
+### Default Cost Functions
+
+Based on results from the [clarity-benchmarking](https://github.com/blockstack/clarity-benchmarking) project, we propose
+new default cost functions. The new costs are supplied in the form of a new Clarity smart contract in [Appendix A](#appendix-a).
+
+### Block Limit Changes
+
+In addition to the runtime cost changes, we propose increasing the
+block limits for MARF reads and writes. Based on the expected performance
+improvements in the implementation of the MARF (see [issue #2869](https://github.com/blockstack/stacks-blockchain/issues/2869))
+we propose doubling the current limits on blocks:
+
+```rust
+pub const BLOCK_LIMIT_MAINNET: ExecutionCost = ExecutionCost {
+    write_length: 15_000_000, // roughly 15 mb
+    write_count: 15_500,
+    read_length: 100_000_000,
+    read_count: 15_500,
+    runtime: 5_000_000_000,
+};
+```
+
+### Changes to Static vs. Dynamic Tabulation of Costs
+
+The cost assessment in Clarity for most data-handling functions (e.g.,
+`map-get?`) use the static cost of the fetch rather than the dynamic
+cost. For more information, see [issue #2864](https://github.com/blockstack/stacks-blockchain/issues/2864)
```

Comment:
Can you expand on the problem in this text?  It's fine to link to github issues for supplementary information, but the SIP should be self-contained.  A reader should be able to read this SIP, and only this SIP, to determine what is meant by "static cost of the fetch rather than the dynamic cost."